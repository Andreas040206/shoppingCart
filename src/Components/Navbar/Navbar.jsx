import NavbarStyles from "./Navbar.module.css";
import LogoImgSrc from "../../assets/Logo.png";
import PropTypes from "prop-types";
import bagIconSrc from "../../assets/bagIcon.png";

const NavbarBtn = ({ text, currentPage, setPage }) => {
  return (
    <button
      onClick={() => setPage(text)}
      className={
        NavbarStyles.menuBtn +
        // If the page is the selected page change the style
        (currentPage === text ? " " + NavbarStyles.menuBtnSelected : "")
      }
    >
      {text}
    </button>
  );
};

NavbarBtn.propTypes = {
  text: PropTypes.string.isRequired,
  currentPage: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  setPage: PropTypes.func.isRequired,
};

const BagItemIcon = ({ shoppingCart }) => {
  const items = shoppingCart.reduce((prev, cur) => {
    return cur.amount + prev;
  }, 0);

  return <div className={NavbarStyles.bagIconCon}>{items}</div>;
};

BagItemIcon.propTypes = {
  shoppingCart: PropTypes.array,
};

export const Navbar = ({ currentPage, setPage, shoppingCart }) => {
  return (
    <div className={NavbarStyles.con}>
      <img src={LogoImgSrc} className={NavbarStyles.logo}></img>
      <div className={NavbarStyles.buttonCon}>
        <NavbarBtn text={"HOME"} currentPage={currentPage} setPage={setPage} />
        <NavbarBtn text={"SHOP"} currentPage={currentPage} setPage={setPage} />
        <NavbarBtn text={"MORE"} currentPage={currentPage} setPage={setPage} />
      </div>
      <div className={NavbarStyles.bagCon}>
        <img className={NavbarStyles.bagIcon} src={bagIconSrc} />
        {shoppingCart.length !== 0 ? (
          <BagItemIcon shoppingCart={shoppingCart} />
        ) : null}
      </div>
    </div>
  );
};

Navbar.propTypes = {
  currentPage: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  setPage: PropTypes.func.isRequired,
  shoppingCart: PropTypes.array,
};
