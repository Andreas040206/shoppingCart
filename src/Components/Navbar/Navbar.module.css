.con {
  height: 60px;
  background-color: rgb(68, 0, 0);
  display: flex;
  justify-content: center;
  align-items: center;
  position: sticky;
  z-index: 20;
  top: 0px;
}

.buttonCon {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 40px;
}

.logo {
  position: absolute;
  left: 20px;
  height: 40px;
}

.bagCon {
  position: absolute;
  right: 30px;
  cursor: pointer;
}

.bagIcon {
  height: 35px;
  transition: 0.3s;
  filter: invert(77%) sepia(13%) saturate(1705%) hue-rotate(327deg)
    brightness(103%) contrast(101%);
}

.bagIcon:hover {
  filter: invert(100%) sepia(95%) saturate(21%) hue-rotate(280deg)
    brightness(106%) contrast(106%);
}

.menuBtn {
  background-color: rgba(0, 0, 0, 0);
  border: none;
  color: rgb(255, 181, 131);
  font-size: 20px;
  transition: 0.3s;
  cursor: pointer;
  text-decoration: none;
}

.menuBtn:hover {
  text-decoration: underline;
  text-underline-offset: 2px;
}

.menuBtnSelected {
  color: white;
}

.bagIconCon {
  position: absolute;
  top: -2px;
  right: -5px;

  height: 20px;
  width: 20px;
  border-radius: 10px;
  background-color: red;
  color: white;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 14px;
  font-weight: 700;
}
