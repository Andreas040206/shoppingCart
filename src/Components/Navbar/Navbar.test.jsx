// From libraryes
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

// Other imports
import { Navbar } from "./Navbar.jsx";
import NavbarStyles from "./Navbar.module.css";

describe("Navbar tests", () => {
  it("Renders the navbar component", () => {
    const { container } = render(<Navbar shoppingCart={[]} />);
    expect(container).toMatchSnapshot();
  });
  it("Change the 'page' value by running set pageFn on click", async () => {
    const user = userEvent.setup();

    let currentPage = "HOME";

    const setFn = (value) => {
      currentPage = value;
    };

    render(
      <Navbar currentPage={currentPage} setPage={setFn} shoppingCart={[]} />
    );
    const shopBtn = screen.getByRole("button", { name: "SHOP" });

    await user.click(shopBtn);

    expect(currentPage).toBe("SHOP");
  });
  it("Gives navbar btns the right clases", () => {
    render(
      <Navbar currentPage={"SHOP"} setPage={() => {}} shoppingCart={[]} />
    );

    const shopBtn = screen.getByRole("button", { name: "SHOP" });
    const homeBtn = screen.getByRole("button", { name: "HOME" });

    expect(homeBtn).not.toHaveClass(NavbarStyles.menuBtnSelected);
    expect(shopBtn).toHaveClass(NavbarStyles.menuBtnSelected);
  });
});
