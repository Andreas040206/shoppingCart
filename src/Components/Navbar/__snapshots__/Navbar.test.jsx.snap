// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar tests > Renders the navbar component 1`] = `
<div>
  <div
    class="_con_7e3bfa"
  >
    <img
      class="_logo_7e3bfa"
      src="/src/assets/Logo.png"
    />
    <div
      class="_buttonCon_7e3bfa"
    >
      <button
        class="_menuBtn_7e3bfa"
      >
        HOME
      </button>
      <button
        class="_menuBtn_7e3bfa"
      >
        SHOP
      </button>
      <button
        class="_menuBtn_7e3bfa"
      >
        MORE
      </button>
    </div>
    <div
      class="_bagCon_7e3bfa"
    >
      <img
        class="_bagIcon_7e3bfa"
        src="/src/assets/bagIcon.png"
      />
    </div>
  </div>
</div>
`;
